ext/src/http/client/curl/http_operation_curl.cc

```diff
@@ -398,61 +401,506 @@ void HttpOperation::Cleanup()
+
+    if (!ssl_options_.ssl_cipher.empty())
+    {
+      /* TLS 1.0, 1.1, 1.2 */
+      const char *cipher_list = ssl_options_.ssl_cipher.c_str();
+
+      rc = SetCurlOption(CURLOPT_SSL_CIPHER_LIST, cipher_list);
+      if (rc != CURLE_OK)
+      {
+        return rc;
+      }
+    }
+
+    if (!ssl_options_.ssl_cipher_suite.empty())
+    {
+#if LIBCURL_VERSION_NUM >= 0x073D00
+      /* TLS 1.3 */
+      const char *cipher_list = ssl_options_.ssl_cipher_suite.c_str();
+
+      rc = SetCurlOption(CURLOPT_TLS13_CIPHERS, cipher_list);
+#else
+      // CURL 7.61.0 (0x07 0x3D 0x00) required for CURLOPT_TLS13_CIPHERS.
+      OTEL_INTERNAL_LOG_ERROR("CURL 7.61.0 required for CIPHER SUITE");
+      return CURLE_UNKNOWN_OPTION;
+#endif
+
+      if (rc != CURLE_OK)
+      {
+        return rc;
+      }
+    }
+
+    if (ssl_options_.ssl_insecure_skip_verify)
+    {
+      /* 6 - DO NOT ENFORCE VERIFICATION, This is not secure. */
+      rc = SetCurlOption(CURLOPT_USE_SSL, (long)CURLUSESSL_NONE);
+      if (rc != CURLE_OK)
+      {
+        return rc;
+      }
+
+      rc = SetCurlOption(CURLOPT_SSL_VERIFYPEER, 0L);
+      if (rc != CURLE_OK)
+      {
+        return rc;
+      }
+
+      rc = SetCurlOption(CURLOPT_SSL_VERIFYHOST, 0L);
+      if (rc != CURLE_OK)
+      {
+        return rc;
+      }
+    }
+    else
+    {
+      /* 6 - ENFORCE VERIFICATION */
+      rc = SetCurlOption(CURLOPT_USE_SSL, (long)CURLUSESSL_ALL);
+      if (rc != CURLE_OK)
+      {
+        return rc;
+      }
+
+      rc = SetCurlOption(CURLOPT_SSL_VERIFYPEER, 1L);
+      if (rc != CURLE_OK)
+      {
+        return rc;
+      }
+
+      rc = SetCurlOption(CURLOPT_SSL_VERIFYHOST, 2L);
+      if (rc != CURLE_OK)
+      {
+        return rc;
+      }
+    }
+  }
+  else
+  {
+    rc = SetCurlOption(CURLOPT_SSL_VERIFYPEER, 0L);
+    if (rc != CURLE_OK)
+    {
+      return rc;
+    }
+
+    rc = SetCurlOption(CURLOPT_SSL_VERIFYHOST, 0L);
+    if (rc != CURLE_OK)
+    {
+      return rc;
+    }
+  }
 
   if (curl_resource_.headers_chunk != nullptr)
   {
-    curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_HTTPHEADER, curl_resource_.headers_chunk);
+    rc = SetCurlOption(CURLOPT_HTTPHEADER, curl_resource_.headers_chunk);
+    if (rc != CURLE_OK)
+    {
+      return rc;
+    }
   }
 
   // TODO: control local port to use
   // curl_easy_setopt(curl, CURLOPT_LOCALPORT, dcf_port);
 
-  curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_TIMEOUT_MS, http_conn_timeout_.count());
+  rc = SetCurlOption(CURLOPT_TIMEOUT_MS, http_conn_timeout_.count());
+  if (rc != CURLE_OK)
+  {
+    return rc;
+  }
 
   // abort if slower than 4kb/sec during 30 seconds
-  curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_LOW_SPEED_TIME, 30L);
-  curl_easy_setopt(curl_resource_.easy_handle, CURLOPT_LOW_SPEED_LIMIT, 4096);
+  rc = SetCurlOption(CURLOPT_LOW_SPEED_TIME, 30L);
+  if (rc != CURLE_OK)
+  {
+    return rc;
+  }
+
+  rc = SetCurlOption(CURLOPT_LOW_SPEED_LIMIT, 4096);
```

Comment:
Should it be `SetCurlOption(CURLOPT_LOW_SPEED_LIMIT, 4096L)` ?

Reply:
Fixed to SetCurlLongOption().